File: src/Logger.rb
module NeoRPGServer
    module Logger
        #------------------------------------------------#
        # Prints out message and puts it in file.
        # Requires: 
        #   prefix - String containing prefix of script/worker.
        #   message - String containing message.
        #------------------------------------------------#
        def self.Log(prefix, message)
            puts "[ ] [#{prefix}] #{message}"
        end
        #------------------------------------------------#
        # Prints out success message and puts it in file.
        # Requires: 
        #   prefix - String containing prefix of script/worker.
        #   successmsg - String containing message.
        #------------------------------------------------#
        def self.LogSuccess(prefix, successmsg)
            puts "[+] [#{prefix}] #{successmsg}"
        end
        #------------------------------------------------#
        # Prints out error message, puts it in file(main.log & error.log) and prints out it in Discord.
        # Requires: 
        #   prefix - String containing prefix of script/worker.
        #   error - String containing error message.
        #------------------------------------------------#
        def self.LogError(prefix, error)
            puts "[-] [#{prefix}] #{error}"
        end
    end
end

File: src/Logger.cs
using System;
using System.Net;
using System.Net.Sockets;

namespace Trifolium {
    public class Logger {
        public static void Log(string prefix, string message) {
            Console.WriteLine("[{0}] {1}", prefix, message);
        }
        public static void LogSuccess(string prefix, string message) {
            Console.ForegroundColor = ConsoleColor.Green;
            Console.WriteLine("[{0}] {1}", prefix, message);
            Console.ForegroundColor = ConsoleColor.White;
        }

        public static void LogError(string prefix, string message) {
            Console.ForegroundColor = ConsoleColor.Red;
            Console.WriteLine("[{0}] {1}", prefix, message);
            Console.ForegroundColor = ConsoleColor.White;
        }
    };
}